reactive/filebeat.py

```diff
@@ -193,7 +196,57 @@ def check_filebeat_repo():
     remove_state("filebeat.repo.changed")
 
 
-@hook("stop")
+@when("apt.installed.filebeat")
+@when('config.changed')
+def sanity_check():
+    """Sanity check config options."""
+    config_items = {
+        'ignore_older': config().get('ignore_older'),
+        'clean_inactive': config().get('clean_inactive')
+    }
+
+    log("Executing sanity checks on config values.")
+
+    # sanity check config strings
+    for cfg, value in config_items.items():
+        if is_valid_time_string(value) is None:
+            log("Invalid config, the value for {} is not an integer."
+                "Passed value: {}.".format(cfg, value))
+            status.blocked("Invalid config. Please check the logs for details.")
+            return
+
+    # sanity check value compliance (e.g., clean_inactive must be greater
+    # than ignore_older + scan_frequency). scan_frequency is no longer a
+    # charm config option but it defaults to 10s and it's set to that in
+    # the charm templates so using that below.
+    scan_frequency = 10
+    ignore_older = int(config_items['ignore_older'])
+    clean_inactive = int(config_items['clean_inactive'])
+
+    if clean_inactive and \
+            int(clean_inactive) <= int(ignore_older) + int(scan_frequency):
```

Comment:
suggestion: no casting needed if the config options are specified to be integers at the charm config level